File: 70-climbing-stairs/70-climbing-stairs.java
class Solution {
    
    public int climbStairs(int n) {
        if(n==1 || n==2)
            return n;
        int[] dp = new int[n+1];
        int lastSecond = 1;
        int last = 2;
        int curr = 0;
        for(int i=3;i<=n;i++){
            curr = last + lastSecond;
            lastSecond = last;
            last = curr;
        }
          
        return last;
    }
}

File: 70-climbing-stairs/70-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        int first = 1;
        if(n==1)
            return first;
        int second = 2;
        if(n==2)
            return second;
        for(int i=3;i<=n;i++){
            int curr = first+second;
            first = second;
            second = curr;
            
        }
        return second;
    }
};